rules/merge.smk

```diff
@@ -10,31 +11,31 @@ rule merge_long:
     output:
         mtx='{results_dir}/summary/{type}/matrix.mtx',
         barcodes='{results_dir}/summary/{type}/barcodes.tsv',
-        features='{results_dir}/summary/{type}/genes.tsv',
+        features='{results_dir}/summary/{type}/features.tsv',
     params:
         samples=lambda wildcards: samples.index
     conda: '../envs/merge_long.yaml'
     script:
         "../scripts/convert_mtx.py"
 
-# rule compress_mtx_summary:
-#     input: 
-#         barcodes='{results_dir}/summary/{type}/barcodes.tsv',
-#         features='{results_dir}/summary/{type}/features.tsv',
-#         mtx='{results_dir}/summary/{type}/matrix.mtx'
-#     output:
-#         barcodes='{results_dir}/summary/{type}/barcodes.tsv.gz',
-#         features='{results_dir}/summary/{type}/features.tsv.gz',
-#         mtx='{results_dir}/summary/{type}/matrix.mtx.gz'
-#     conda: '../envs/pigz.yaml'
-#     threads: 3
-#     shell:
-#         """pigz -p {threads} {input.barcodes} {input.features} {input.mtx}"""
+rule compress_mtx_summary:
+    input: 
+        barcodes='{results_dir}/summary/barcodes.tsv',
+        features='{results_dir}/summary/features.tsv',
+        mtx='{results_dir}/summary/matrix.mtx'
+    output:
+        barcodes='{results_dir}/summary/barcodes.tsv.gz',
+        features='{results_dir}/summary/features.tsv.gz',
+        mtx='{results_dir}/summary/matrix.mtx.gz'
+    conda: '../envs/pigz.yaml'
+    threads: 3
+    shell:
+        """pigz -p {threads} {input.barcodes} {input.features} {input.mtx}"""
 
 rule violine_plots:
     input:
-        umi_mtx='{results_dir}/summary/umi/matrix.mtx',
-        read_mtx='{results_dir}/summary/umi/matrix.mtx',
+        umi_mtx='{results_dir}/summary/umi/matrix.mtx.gz',
+        read_mtx='{results_dir}/summary/umi/matrix.mtx.gz',
```

Comment:
This is why the compress_mtx_summary wasn't running on umi and reads. This should be:  `read_mtx='{results_dir}/summary/read/matrix.mtx.gz'`

This will allow you to not duplicate the compressing of summaries

Reply:
Ah, that’s less messy.